server.py

```diff
@@ -1,58 +1,78 @@
-import threading
-from flask import Flask
-from flask_socketio import SocketIO, emit
-from canvas_board import CanvasBoard, CanvasBoardEncoder
+import random, string
+from flask import Flask, jsonify
+from flask_socketio import SocketIO, emit, join_room, leave_room
+from canvas_board import CanvasBoard
+from json import dumps
 
 # constants
 PORT = 8080
 HEIGHT = 500
 WIDTH = 500
+ID_LENGTH = 7
 
 # create Flask object
-app = Flask(__name__)
+app = Flask(__name__, template_folder='static/', static_folder='static/')
 app.config['SECRET_KEY'] = 'secret!'
 socketio = SocketIO(app, cors_allowed_origins='*')
 
 # server class
 class Server:
-    def __init__(self, imagedata):
-        self.board = CanvasBoard(imagedata)
-        self.lock = threading.Lock()
-
-    def update_board(self, diffs: list):
-        self.lock.acquire()
-        self.board.update_board(diffs)
-        self.lock.release()
+    def __init__(self, boards: 'dict that maps room id to CanvasBoard'):
+        self.boards = boards

```

Comment:
Since it's possible that the server starts without any rooms, instead of needing to initialize the server with a dict of boards, we can just initialize an empty dict.

We should expose a method to add a new room though (and retrieve the board for a room).